views/chunks.sql

```diff
@@ -0,0 +1,43 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE and LICENSE for copyright and licensing information.
+
+DROP VIEW IF EXISTS chunks_ts, chunks_tstz;
+
+-- Show time range, hypertable, and tablespace for chunks belonging to
+-- hypertables that have a time dimension.
+--
+-- This do not show chunks belonging to hypertables that use an
+-- integer time dimension.
+CREATE OR REPLACE VIEW chunks_tstz AS
+SELECT format('%1$I.%2$I', ch.schema_name, ch.table_name)::regclass AS chunk
+     , format('%1$I.%2$I', ht.schema_name, ht.table_name)::regclass AS hypertable
+     , tstzrange(CASE WHEN sl.range_start = -9223372036854775808 THEN NULL
+                 ELSE TIMESTAMPTZ 'epoch' + sl.range_start * INTERVAL '1 microsecond' END,
+                 CASE WHEN sl.range_end = 9223372036854775807 THEN NULL
+                 ELSE TIMESTAMPTZ 'epoch' + sl.range_end * INTERVAL '1 microsecond' END)
```

Comment:
why aren't we using the _timescaledb_internal function that converts to/from our internal representation?

Reply:
Not sure which internal function you're referring to. `range_value_to_pretty`? That function returns a text rather than a timestamp. I could, however, create a convenience function that does the above job and return the proper type. Something like

```sql
CREATE FUNCTION range_to_type(BIGINT, base "anyelement")
RETURNS "anyelement" AS $$
BEGIN
  RETURN CASE $1 
    WHEN -9223372036854775808 THEN NULL
    WHEN 9223372036854775807 THEN NULL
    ELSE base + $1 * INTERVAL '1 microsecond'
  END;
END $$ LANGUAGE plpgSQL;
```

Reply:
From util_time.sql 
```
-- This file contains utilities for time conversion.
CREATE OR REPLACE FUNCTION _timescaledb_internal.to_unix_microseconds(ts TIMESTAMPTZ) RETURNS BIGINT
    AS '@MODULE_PATHNAME@', 'ts_pg_timestamp_to_unix_microseconds' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;

CREATE OR REPLACE FUNCTION _timescaledb_internal.to_timestamp(unixtime_us BIGINT) RETURNS TIMESTAMPTZ
    AS '@MODULE_PATHNAME@', 'ts_pg_unix_microseconds_to_timestamp' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;

CREATE OR REPLACE FUNCTION _timescaledb_internal.to_timestamp_without_timezone(unixtime_us BIGINT)
  RETURNS TIMESTAMP
  AS '@MODULE_PATHNAME@', 'ts_pg_unix_microseconds_to_timestamp'
  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;

CREATE OR REPLACE FUNCTION _timescaledb_internal.to_date(unixtime_us BIGINT)
  RETURNS DATE
  AS '@MODULE_PATHNAME@', 'ts_pg_unix_microseconds_to_date'
  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;

CREATE OR REPLACE FUNCTION _timescaledb_internal.to_interval(unixtime_us BIGINT) RETURNS INTERVAL
    AS '@MODULE_PATHNAME@', 'ts_pg_unix_microseconds_to_interval' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
```

Reply:
I think those will do most of this work and we should probably just have one way of doing it.

Reply:
They do indeed, so I replaced the addition with a call to these functions.

A general observation: the functions do not return NULL when passed the outer range of the value (what we internally use to denote infinity), which seems kind of strange and make usage more complicated than necessary (see the the `timescaledb_information.continuous_aggregates` for example).

Reply:
do we have infinite ranges in any of our chunks anyway?

Reply:
No, but in some cases we need to do additional checks to handle the "infinite" case, which would be better to handle as NULL rather than throw an error if the user makes a mistake.